transfocate/lens.py

```diff
@@ -5,133 +5,91 @@
 # Standard #
 ############
 import logging
+
 ###############
 # Third Party #
 ###############
-from ophyd import Device, EpicsSignal, EpicsSignalRO
-from ophyd import Component
-from ophyd.utils import set_and_wait
-logger = logging.getLogger(__name__)
+import numpy as np
+import prettytable
+from ophyd import EpicsSignalRO
+from ophyd import FormattedComponent as FC
+from pcdsdevices.inout import InOutRecordPositioner
 
 ##########
 # Module #
 ##########
 
-class Lens(Device):
+logger = logging.getLogger(__name__)
+
+
+class Lens(InOutRecordPositioner):
     """
     Data structure for basic Lens object
-    """
-    
-    """
+
     Parameters
     ----------
-    sig_radius : EPICS Read Only signal
-        Radius of beryllium lens measured in microns (um). Affects focus of lens  
-    sig_z : EPICS Read Only signal
-        Lens position along beam pipelin measure in meters (m).
-    sig_focus : EPICS Read Only signal
-        Focal length of lens in meters (m). Is a function of radius
-    state : EPICS Read Only signal
-        Position of the lens.  1 if it is inserted in the beamline and 0 if
-        it is removed
-    in_signal : EPICS signal
-        Signal that triggers EPICS to insert the lens from the beamline
-    out_signal : EPICS signal
-        Signal that triggers EPICS to remove the lens from the beamline
+    prefix : str
+        Name of the state record that controls the PV
 
-    Note
-    ----
-    The variables radius, z, and focus are now EPICS signals for the lenses.
+    prefix_lens : str
+        Prefix for the PVs that contain focusing information
     """
-    #defining the EPICS variables (note: these are the signals and not the
-    #values of each variable)
-    sig_radius=Component(EpicsSignalRO, "RADIUS", auto_monitor=True)
-    sig_z = Component(EpicsSignalRO, "Z", auto_monitor=True)
-    sig_focus = Component(EpicsSignalRO, "FOCUS", auto_monitor=True)
-    state = Component(EpicsSignalRO, "STATE") 
-    in_signal = Component(EpicsSignal, "INSERT")
-    out_signal = Component(EpicsSignal, "REMOVE")
+    _sig_radius = FC(EpicsSignalRO, "{self.prefix_lens}:RADIUS",
+                     auto_monitor=True)
+    _sig_z = FC(EpicsSignalRO, "{self.prefix_lens}:Z",
+                auto_monitor=True)
+    _sig_focus = FC(EpicsSignalRO, "{self.prefix_lens}:FOCUS",
+                    auto_monitor=True)
```

Comment:
Why are these all preceded by underscores?

Reply:
I would to avoid confusing users with both properties and signals. This way there is no confusion between `sig_focus` and `focus`